From 163752f4b129cb8e61bff765bccbac5990b97507 Mon Sep 17 00:00:00 2001
From: Daniel Steger <daniel.steger@xilinx.com>
Date: Mon, 25 Feb 2019 10:59:58 -0800
Subject: [PATCH 0523/1852] media: xilinx: tpg: dt pixel per clock validate.

commit 70d227db755311c796044d9bf629401389821182 from
https://github.com/Xilinx/linux-xlnx.git

This patch fixes a bug where the driver rejects a pixel per clock
setting of 1. This is incorrect behavior because the tpg ip supports
1,2,4,8 pixel per clock settings.

Signed-off-by: Daniel Steger <daniel.steger@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-tpg.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-tpg.c b/drivers/media/platform/xilinx/xilinx-tpg.c
index 99b9fb6233d7..b88befa5357d 100644
--- a/drivers/media/platform/xilinx/xilinx-tpg.c
+++ b/drivers/media/platform/xilinx/xilinx-tpg.c
@@ -104,7 +104,6 @@
 #define XTPG_MAX_HEIGHT			(7760)
 
 #define XTPG_MIN_PPC			1
-#define XTPG_MAX_PPC			8
 
 #define XTPG_MIN_FRM_INT		1
 
@@ -976,9 +975,8 @@ static int xtpg_parse_of(struct xtpg_device *xtpg)
 	if (ret < 0) {
 		xtpg->ppc = XTPG_MIN_PPC;
 		dev_dbg(dev, "failed to read ppc in dt\n");
-	} else if ((xtpg->ppc > XTPG_MAX_PPC)
-		|| (xtpg->ppc < XTPG_MIN_PPC)
-		|| (xtpg->ppc % 2)) {
+	} else if ((xtpg->ppc != 1) && (xtpg->ppc != 2) &&
+			(xtpg->ppc != 4) && (xtpg->ppc != 8)) {
 		dev_err(dev, "Invalid ppc config in dt\n");
 		return -EINVAL;
 	}
-- 
2.31.1

